### ISSUE TITLE
aria-describedby on input type file are not announced

### ISSUE THREAD

**Issue by lallysmbc** [nvaccess/nvda#11742]
### Steps to reproduce:
When having a html input of type file with an associated aria-describedby property the text associated is not Announced or displayed within the screen viewer on tabbing.

**NOTE:** This appears to only be firefox, chrome correctly reads out the associated aria-describedby

url: https://design-system.service.gov.uk/components/file-upload/error/index.html

1) Visit the govuk file upload example above with an associated aria-describedby atttribute
2) tab into the input
3) the associated aria-describedby text is not announced or displayed within the speech viewer

### Actual behavior:

1) Visit the govuk file upload example above with an associated aria-describedby atttribute
2) tab into the input
3) the associated aria-describedby text is not announced or displayed within the speech viewer

### Expected behavior:

1) Visit the govuk file upload example above with an associated aria-describedby atttribute
2) tab into the input
3) the associated aria-describedby text should be announced and displayed within the speech viewer

Associated Html structure
```
<div class="govuk-form-group govuk-form-group--error">
  <label class="govuk-label" for="file-upload-1">
    Upload a file
  </label>
  <span id="file-upload-1-error" class="govuk-error-message">
  <span class="govuk-visually-hidden">Error:</span> The CSV must be smaller than 2MB
  </span>
  <input class="govuk-file-upload govuk-file-upload--error" id="file-upload-1" name="file-upload-1" type="file" aria-describedby="file-upload-1-error">
</div>
```

#### NVDA installed/portable/running from source:
installed

#### NVDA version:
2020.2

#### Windows version:
10 Build 19041

#### Name and version of other software in use when reproducing the issue:
firefox 81.0.1 (64-bit)

#### Does the issue still occur after restarting your computer?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no

#### If addons are disabled, is your problem still occuring?
no extension installed

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
no

**Comment by mmerezhko-hv** [comment]
Same for me

**Comment by terohotti** [comment]
Just noticed the same.  FF 95.0.2 NVDA 2021.2. 

Chrome working fine!

**Comment by Adriani90** [comment]
This is a different approach used in Firefox I guess. The aria_describedby is announced only when landing with the focus on the button in focus mode. In browse mode however, the aria_describedby is announced only when using the arrow keys to navigate, and not when pressing b or tab or any other command to focus the button. For the screen reader it appears as if the description is directly above the button.

cc: @jcsteh

**Comment by zoltan-dulac** [comment]
This is not a different approach on Firefox.  If using focus mode on [my codepen demo](https://codepen.io/zoltan-dulac/pen/ZEVbaXP), it doesn't announce the aria-describedby on the file input field.  This doesn't happen on any other form element besides <input type="file">

Is there any way this can be fixed?  This is a pretty serious issue.

**Comment by jcsteh** [comment]
When I tab to the button in your Codepen in focus mode, I get this:
Upload document: grouping Must be less that 5MB in size
Browse... button
It doesn't work in browse mode though because NVDA doesn't report the description of groupings in browse mode.

**Comment by Adriani90** [comment]
> It doesn't work in browse mode though because NVDA doesn't report the description of groupings in browse mode.

What is the reason behind this? Could NVDA at least announce "has description" so that we can envoke it by pressing e.g. nvda+d? This could apply for both focus and browse mode and would probably reduce noisy verbosity. Ideally it could be made configurable so an user can decide to hear the description while navigating or when pressing the nvda+d command.

**Comment by jcsteh** [comment]
In Firefox 121 (currently beta), file input accessibility has been changed so that it is a single button rather than a grouping, since multiple screen readers (including NVDA) don't handle the grouping correctly. See https://bugzilla.mozilla.org/show_bug.cgi?id=1661923.
